pg_backup_api/requirements.txt

```diff
@@ -1,12 +1,30 @@
-barman >= 2.16
-connexion[swagger-ui] >= 2.6.0; python_version>="3.6"
-# 2.3 is the last version that supports python 3.4-3.5
-connexion[swagger-ui] <= 2.3.0; python_version=="3.5" or python_version=="3.4"
-# connexion requires werkzeug but connexion < 2.4.0 does not install werkzeug
-# we must peg werkzeug versions below to fix connexion
-# https://github.com/zalando/connexion/pull/1044
-werkzeug == 0.16.1; python_version=="3.5" or python_version=="3.4"
-swagger-ui-bundle >= 0.0.2
-python_dateutil >= 2.6.0
-setuptools >= 21.0.0
-Flask == 1.1.2
+argcomplete==2.0.0
+attrs==21.4.0
+barman>=2.17
+certifi==2021.10.8
+charset-normalizer==2.0.10
+click==8.0.3
+clickclick==20.10.2
+connexion==2.10.0
+dataclasses>=0.6
+Flask==2.0.2
+idna==3.3
+importlib-metadata==4.8.3
+inflection==0.5.1
+isodate==0.6.1
+itsdangerous==2.0.1
+Jinja2==3.0.3
+jsonschema==3.2.0
+MarkupSafe==2.0.1
+openapi-schema-validator==0.1.6
+openapi-spec-validator==0.3.2
+psycopg2==2.9.3
+pyrsistent==0.16.1
+python-dateutil==2.8.2
+PyYAML==5.4.1
+requests==2.27.1
+six==1.16.0
+typing-extensions==4.0.1
+urllib3==1.26.8
+Werkzeug==2.0.2
+zipp==3.6.0
```

Comment:
This looks like a lot of dependencies. did you freeze all environment versions? If so was it on purpose?

Reply:
Yes we had to freeze the entire virtualenv resulting from installing the top level deps in `setup.py` so that fpm consistently creates the same virtualenv each time it runs.